### PR TITLE
ScreenshotManager: Fix saving to clipboard

### PR DESCRIPTION
**Pull request by lenemter**
Addresses https://github.com/elementary/gala/issues/2014

### CODE REVIEW

File: vapi/libmutter.vapi [hunk 1]
```diff
@@ -793,6 +793,7 @@ namespace Meta {
 	public sealed class SelectionSourceMemory : Meta.SelectionSource {
 		[CCode (has_construct_function = false, type = "MetaSelectionSource*")]
 		public SelectionSourceMemory (string mimetype, GLib.Bytes content);
```
Comment (leolost2605) [inline vapi/libmutter.vapi:795]: AFAIK you can just do something like this:
```suggestion
		public SelectionSourceMemory (string mimetype, GLib.Bytes content) throws GLib.Error;
```
which avoids having to use the static function

File: src/ScreenshotManager.vala [hunk 2]
```diff
@@ -342,19 +342,40 @@ namespace Gala {
...
-            unowned Gtk.Clipboard clipboard = Gtk.Clipboard.get_default (display);
-
             var screenshot = Gdk.pixbuf_get_from_surface (image, 0, 0, image.get_width (), image.get_height ());
             if (screenshot == null) {
-                warning ("could not save screenshot to clipboard: null pixbuf");
+                warning ("Could not save screenshot to clipboard: null pixbuf");
+                return false;
+            }
+
+            var stream = new GLib.MemoryOutputStream.resizable ();
```
Comment (leolost2605) [inline src/ScreenshotManager.vala:354]: I'm really not to knowledgeable in this area so what's the advantage of using a memory stream here? Wouldn't `save_to_buffer` together with `new Bytes.take ()` do the same? Or is saving to a memory stream here somehow faster?

**Review by leolost2605 — CHANGES_REQUESTED** [review]

File: vapi/libmutter.vapi (hunk 1 above)
Comment (lenemter) [reply vapi/libmutter.vapi]: It didn't work for me when I tried it, but now for some reason it just works

File: src/ScreenshotManager.vala (hunk 2 above)
Comment (lenemter) [reply src/ScreenshotManager.vala]: I used it because gnome-shell uses it, `GLib.Bytes.take` works just fine. Thank you.

**Review by leolost2605 — APPROVED** [review]
Very cool :)

**Pull request merged**